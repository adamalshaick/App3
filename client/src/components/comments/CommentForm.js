import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import TextAreaFieldGroup from "../common/TextAreaFieldGroup";
import { addComment } from "../../actions/postActions";

class CommentForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: ""
    };
  }

  componentWillReceiveProps(newProps) {
    if (newProps.errors) {
      this.setState({ errors: newProps.errors });
    }
  }

  onSubmit = e => {
    e.preventDefault();

    const { user } = this.props.auth;
    const { postId } = this.props;

    const newComment = {
      text: this.state.text,
      name: user.name,
      avatar: user.avatar
    };

    this.props.addComment(postId, newComment);
    this.setState({ text: "" });
  };

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <div className="col-12">
        <form onSubmit={this.onSubmit}>
          <TextAreaFieldGroup
            className="form-control"
            placeholder="Make a comment..."
            name="text"
            value={this.state.text}
            onChange={this.onChange}
            style={{ fontSize: "12px", width: "100%" }}
          />

          <div className="float-right">
            <button type="submit" className="btn btn-outline-primary btn-sm">
              Submit
            </button>
          </div>
        </form>
      </div>
    );
  }
}

CommentForm.propTypes = {
  addPost: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  postId: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { addComment }
)(CommentForm);
