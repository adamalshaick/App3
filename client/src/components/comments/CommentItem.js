import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { deleteComment } from "../../actions/postActions";
import Link from "react-router-dom/Link";

class CommentItem extends Component {
  onDeleteClick(postId, commentId) {
    this.props.deleteComment(postId, commentId);
  }

  render() {
    const { comment, postId, auth } = this.props;
    return (
      <div className=" mb-3">
        <div className="row mt-3">
          <Link to="">
            <img
              style={{ width: "40px" }}
              className="rounded-circle  ml-4"
              src={`../uploads/post_image/${profile.profileImage}`}
              alt=""
            />
          </Link>

          <div className="text-center mt-2 ml-2">{comment.name}</div>

          <div className="col-12">
            <div style={{ fontSize: "0.9rem" }} className="float-left ml-2">
              {comment.text}
            </div>
          </div>

          {comment.user === auth.user.id ? (
            <button
              onClick={this.onDeleteClick.bind(this, postId, comment._id)}
              type="button"
              className="btn btn-outline-danger btn-sm  mr-1"
            >
              Delete
            </button>
          ) : null}
        </div>
      </div>
    );
  }
}

CommentItem.propTypes = {
  deleteComment: PropTypes.func.isRequired,
  comment: PropTypes.object.isRequired,
  postId: PropTypes.string.isRequired,
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { deleteComment }
)(CommentItem);
