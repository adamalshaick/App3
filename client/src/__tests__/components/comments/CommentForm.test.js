import React from "react";
import { shallow } from "enzyme";
import { CommentForm } from "../../../components/comments/CommentForm";
import toJson from "enzyme-to-json";

const mockAddComment = jest.fn();

const Props = {
  currentProfile: {
    user: {
      name: "testName"
    },
    profileImage: "testAvatar"
  },
  postId: "postId"
};

const wrapper = shallow(<CommentForm {...Props} addComment={mockAddComment} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("add comment action", () => {
  it("calls function", () => {
    wrapper
      .find("#commentInput")
      .simulate("change", { target: { name: "text", value: "test comment" } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(mockAddComment.mock.calls).toEqual([
      [
        "postId",
        { avatar: "testAvatar", name: "testName", text: "test comment" }
      ]
    ]);
  });
});
