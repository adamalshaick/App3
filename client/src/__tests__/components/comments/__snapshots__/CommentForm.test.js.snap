// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="col-12 entry"
>
  <form
    onSubmit={[Function]}
  >
    <TextAreaFieldGroup
      className="form-control"
      id="commentInput"
      name="text"
      onChange={[Function]}
      placeholder="Make a comment..."
      style={
        Object {
          "fontSize": "12px",
          "width": "100%",
        }
      }
      value=""
    />
    <div
      className="float-right"
    >
      <button
        className="btn btn-outline-primary btn-sm"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
